Tests/Algorithm/Framework/Portfolio/InsightWeightingPortfolioConstructionModelTests.cs

```diff
@@ -13,311 +13,54 @@
-        public void AutomaticallyRemoveInvestedWithNewInsights(Language language, InsightDirection direction)
-        {
-            SetPortfolioConstruction(language, _algorithm);
-
-            // Let's create a position for SPY
-            var insights = new[] { GetInsight(Symbols.SPY, direction, _algorithm.UtcTime) };
-
-            foreach (var target in _algorithm.PortfolioConstruction.CreateTargets(_algorithm, insights))
-            {
-                var holding = _algorithm.Portfolio[target.Symbol];
-                holding.SetHoldings(holding.Price, target.Quantity);
-                _algorithm.Portfolio.SetCash(_startingCash - holding.HoldingsValue);
-            }
-
-            SetUtcTime(_algorithm.UtcTime.AddDays(2));
-
-            var amount = _algorithm.Portfolio.TotalPortfolioValue * (decimal)Weight;
-            var expectedTargets = _algorithm.Securities.Select(x =>
-            {
-                // Expected target quantity for SPY is zero, since it will be removed
-                var quantity = x.Key.Value == "SPY" ? 0 : (int)direction
-                                                          * Math.Floor(amount * (1 - _algorithm.Settings.FreePortfolioValuePercentage)
-                                                                       / x.Value.Price);
-                return new PortfolioTarget(x.Key, quantity);
-            });
-
-            // Do no include SPY in the insights
-            insights = _algorithm.Securities.Keys.Where(x => x.Value != "SPY")
-                .Select(x => GetInsight(x, direction, _algorithm.UtcTime)).ToArray();
+        private const double _weight = 0.01;
 
-            var actualTargets = _algorithm.PortfolioConstruction.CreateTargets(_algorithm, insights);
-
-            AssertTargets(expectedTargets, actualTargets);
-        }
+        public override double? Weight => _weight;
 
         [Test]
         [TestCase(Language.CSharp)]
         [TestCase(Language.Python)]
-        public void AutomaticallyRemoveInvestedWithoutNewInsights(Language language)
+        public void WeightsProportionally(Language language)
         {
-            SetPortfolioConstruction(language, _algorithm);
+            SetPortfolioConstruction(language);
 
-            // Let's create a position for SPY
-            var insights = new[] { GetInsight(Symbols.SPY, InsightDirection.Up, _algorithm.UtcTime) };
-
-            foreach (var target in _algorithm.PortfolioConstruction.CreateTargets(_algorithm, insights))
+            var direction = InsightDirection.Down;
+            if (PortfolioBias != PortfolioBias.LongShort && (int)direction != (int)PortfolioBias)
             {
-                var holding = _algorithm.Portfolio[target.Symbol];
-                holding.SetHoldings(holding.Price, target.Quantity);
-                _algorithm.Portfolio.SetCash(_startingCash - holding.HoldingsValue);
+                direction = InsightDirection.Flat;
```

Comment:
Hm maybe could use `Direction.Up` when LongOnly bias, else this test won't really cover what its meant to? or actually even better directly use `Direction.Up` for all the cases.

Reply:
It is right. When the `PortfolioBias` and `InsightDirection` don't meet (`PortfolioBias.LongOnly` and `InsightDirection.Down`), the `InsightDirection` is `InsightDirection.Flat`. We are creating insights with `InsightDirection.Down` in that test.